jsx/StaticDataTable.js

```diff
@@ -193,14 +267,8 @@ StaticDataTable = React.createClass({
                     data = this.props.Data[index[i].RowIdx][j];
                 }
 
-                // Increase counter, if filter value is found to be a substring
-                // of one of the column values
-                var filterData = this.props.Filter[this.props.Headers[j]];
-                if (filterData !== null &&
-                    data !== null &&
-                    data.indexOf(filterData) > -1
-                ) {
-                    filterMatchCount++;
+                if (this.hasFilterKeyword(this.props.Headers[j], data)) {
```

Comment:
The function compares a specific column, but it's in a loop and there's no break out of the loop, so I'm still not following why the next iteration through the loop won't increase it again when it compares the second column, resulting in the row being counted twice.